sensor/common/scan/scan.go

```diff
@@ -68,22 +74,22 @@ func NewLocalScan(registryStore *registry.Store) *LocalScan {
 }
 
 // EnrichLocalImageInNamespace invokes enrichLocalImageFromRegistry with a slice of credentials from the registryStore based on namespace as well as
-// the OCP global pull secret
+// the OCP global pull secret.
 //
-// If no registry credentials are found an empty registry slice is passed to enrichLocalImageFromRegistry for enriching with 'no auth'
-func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClient v1.ImageServiceClient, ci *storage.ContainerImage, namespace string) (*storage.Image, error) {
-	if namespace == "" {
-		// If no namespace provided try enrichment with 'no auth'
-		return s.enrichLocalImageFromRegistry(ctx, centralClient, ci, nil)
-	}
-
+// If no registry credentials are found an empty registry slice is passed to enrichLocalImageFromRegistry for enriching with 'no auth'.
+func (s *LocalScan) EnrichLocalImageInNamespace(ctx context.Context, centralClient v1.ImageServiceClient, ci *storage.ContainerImage, namespace string, requestID string, force bool) (*storage.Image, error) {
 	var regs []registryTypes.Registry
-
 	var reg registryTypes.Registry
+	var err error
+
 	imgName := ci.GetName()
-	reg, err := s.getRegistryForImageInNamespace(imgName, namespace)
-	if err == nil {
-		regs = append(regs, reg)
+
+	if namespace != "" {
+		// if namespace provided pull appropriate registry.
+		reg, err := s.getRegistryForImageInNamespace(imgName, namespace)
+		if err == nil {
```

Comment:
might have missed this in the other review, but: why do we ignore the error here?

Reply:
The method `getRegistryForImageInNamespace` will only return an error if an entry is not found. So its not being ignored, just the err content itself is not of value, but it's existence is used.